File: microservices/categories-service/src/main/java/com/categoriesservice/categories/repository/CategoryRepository.java
package com.categoriesservice.categories.repository;

import com.categoriesservice.categories.model.Category;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.List;

/**
 * Category Repository - Interfaz para acceso a datos de categorías
 */
@Repository
public interface CategoryRepository extends JpaRepository<Category, Long> {
    
    List<Category> findByUserId(Long userId);
    List<Category> findByNameContainingIgnoreCase(String name);
}

File: microservices/categories-service/src/main/kotlin/com/categoriesservice/categories/repository/CategoryRepository.kt
package com.categoriesservice.categories.repository

import com.categoriesservice.categories.model.Category
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface CategoryRepository : JpaRepository<Category, Long> {
    fun findByActiveTrue(): List<Category>
    fun existsByName(name: String): Boolean
}
